### PR TITLE
[XPTI] Refactoring - Removing TBB dependency

### PR DESCRIPTION
**Pull request by tovinkere**
XPTI framework provided an option to build it with std::thread or TBB, with std::thread being the default library. However, many of the performance and debug tests relied on TBB to check for data races and data consistency. TBB constructs have been removed from the framework and the tests have been reworked to use std::thread

### CODE REVIEW

File: xptifw/CMakeLists.txt [hunk 1]
```diff
@@ -13,8 +13,10 @@ if (NOT DEFINED XPTI_DIR) # don't overwrite if already set
   if (NOT DEFINED XPTI_SOURCE_DIR)
     # If XPTI_SOURCE_DIR is not provided then assume that xpti is in-tree and not external.
     set(XPTI_DIR ${CMAKE_CURRENT_LIST_DIR}/../xpti)
+    message(STATUS "Setting XPTI_DIR using relative path...")
   else()
     set(XPTI_DIR ${XPTI_SOURCE_DIR}/../xpti)
+    message(STATUS "Setting XPTI_DIR using XPTI_SOURCE_DIR from command...")
```
Comment (KseniyaTikhomirova) [inline xptifw/CMakeLists.txt:19]: it is left from local debug process or added intentionally?

File: xptifw/basic_test/CMakeLists.txt [hunk 2]
```diff
@@ -6,11 +6,11 @@ add_executable(XPTIFWBasicTests ${SOURCES})
 target_link_libraries(XPTIFWBasicTests PRIVATE xptifw ${CMAKE_DL_LIBS})
 
 if (UNIX)
-  target_compile_definitions(XPTIFWBasicTests PRIVATE -gline-tables-only -O3)
+  add_compile_options(-gline-tables-only -O3)
 endif()
 
 # There's a XPTI_ENABLE_TBB guard in xptifw/CMakeLists.txt
-target_link_libraries(XPTIFWBasicTests PRIVATE tbb)
+#target_link_libraries(XPTIFWBasicTests PRIVATE tbb)
```
Comment (KseniyaTikhomirova) [inline xptifw/basic_test/CMakeLists.txt:13]: should it be totally removed?

File: xptifw/basic_test/CMakeLists.txt [hunk 3]
```diff
@@ -6,11 +6,11 @@ add_executable(XPTIFWBasicTests ${SOURCES})
 target_link_libraries(XPTIFWBasicTests PRIVATE xptifw ${CMAKE_DL_LIBS})
 
 if (UNIX)
-  target_compile_definitions(XPTIFWBasicTests PRIVATE -gline-tables-only -O3)
+  add_compile_options(-gline-tables-only -O3)
```
Comment (KseniyaTikhomirova) [inline xptifw/basic_test/CMakeLists.txt:9]: could you please clarify what this change stands for?

File: xptifw/CMakeLists.txt (hunk 1 above)
Comment (tovinkere) [reply xptifw/CMakeLists.txt:19]: Was left intentionally, but removed it as it adds no value.

File: xptifw/basic_test/CMakeLists.txt (hunk 2 above)
Comment (tovinkere) [reply xptifw/basic_test/CMakeLists.txt:13]: Removed!

File: xptifw/basic_test/CMakeLists.txt (hunk 3 above)
Comment (tovinkere) [reply xptifw/basic_test/CMakeLists.txt:9]: That line used to cause CMake errors and has been changed to ensure we have -O3 for the basic performance tests

**Comment by dm-vodopyanov** [comment]
image

No idea what is that :confused:

@bader @aelovikov-intel FYI, this is strange, no activity from my side in this PR, but Github produced and showed these actions...

**Comment by tovinkere** [comment]
Failed Tests (1):
SYCL :: ESIMD/accessor_local.cpp - Reported in https://github.com/intel/llvm/issues/10138.

**Comment by aelovikov-intel** [comment]
@tovinkere , you don't have code-review approval yet, FYI.

**Review by KseniyaTikhomirova — APPROVED** [review]
LGTM

**Comment by tovinkere** [comment]
> @tovinkere , you don't have code-review approval yet, FYI.

Reviews are done - looks like one of the errors have been previously reported and the other failure is unrelated too. How to proceed for the merge?

**Comment by dm-vodopyanov** [comment]
> > @tovinkere , you don't have code-review approval yet, FYI.
> 
> Reviews are done - looks like one of the errors have been previously reported and the other failure is unrelated too. How to proceed for the merge?

https://github.com/intel/llvm/blob/sycl/CONTRIBUTING.md#merge

**Pull request merged**